File: test/python/vss_test.py
import duckdb
import os
import pytest

# Get a fresh connection to DuckDB with the vss extension binary loaded
@pytest.fixture
def duckdb_conn():
    extension_binary = os.getenv('VSS_EXTENSION_BINARY_PATH')
    if (extension_binary == ''):
        raise Exception('Please make sure the `VSS_EXTENSION_BINARY_PATH` is set to run the python tests')
    conn = duckdb.connect('', config={'allow_unsigned_extensions': 'true'})
    conn.execute(f"load '{extension_binary}'")
    return conn

def test_vss(duckdb_conn):
    duckdb_conn.execute("SELECT vss('Sam') as value;");
    res = duckdb_conn.fetchall()
    assert res[0][0] == "Vss Sam 🐥"

def test_vss_openssl_version_test(duckdb_conn):
    duckdb_conn.execute("SELECT vss_openssl_version('Michael');");
    res = duckdb_conn.fetchall()
    assert res[0][0][0:51] == "Vss Michael, my linked OpenSSL version is OpenSSL"

File: test/nodejs/vss_test.js
var duckdb = require('../../duckdb/tools/nodejs');
var assert = require('assert');

describe(`vss extension`, () => {
    let db;
    let conn;
    before((done) => {
        db = new duckdb.Database(':memory:', {"allow_unsigned_extensions":"true"});
        conn = new duckdb.Connection(db);
        conn.exec(`LOAD '${process.env.VSS_EXTENSION_BINARY_PATH}';`, function (err) {
            if (err) throw err;
            done();
        });
    });

    it('vss function should return expected string', function (done) {
        db.all("SELECT vss('Sam') as value;", function (err, res) {
            if (err) throw err;
            assert.deepEqual(res, [{value: "Vss Sam 🐥"}]);
            done();
        });
    });

    it('vss_openssl_version function should return expected string', function (done) {
        db.all("SELECT vss_openssl_version('Michael') as value;", function (err, res) {
            if (err) throw err;
            assert(res[0].value.startsWith('Vss Michael, my linked OpenSSL version is OpenSSL'));
            done();
        });
    });
});